src/main/scala/com/ubirch/discovery/services/connector/JanusGraphConnector.scala

```diff
@@ -25,57 +25,56 @@ import scala.collection.JavaConverters._
-    conf.addProperty("serializer.className", config.getString("core.connector.serializer.className"))
-    // no idea why the following line needs to be duplicated. Doesn't work without
-    // cf https://stackoverflow.com/questions/45673861/how-can-i-remotely-connect-to-a-janusgraph-server first answer, second comment ¯\_ツ_/¯
-    conf.addProperty("serializer.config.ioRegistries", config.getAnyRef("core.connector.serializer.config.ioRegistries").asInstanceOf[java.util.ArrayList[String]])
-    conf.addProperty("serializer.config.ioRegistries", config.getStringList("core.connector.serializer.config.ioRegistries"))
+    cluster.addContactPoints(hosts: _*)
+      .port(config.getInt("core.connector.port"))
 
+    val maxWaitForConnection = config.getInt("core.connector.connectionPool.maxWaitForConnection")
+    if (maxWaitForConnection > 0) cluster.maxWaitForConnection(maxWaitForConnection)
     Try(config.getInt("settings.connectionPool.maxContentLength")) match {
-      case Success(value) => conf.addProperty("settings.connectionPool.maxContentLength", value)
-      case Failure(_) => conf.addProperty("settings.connectionPool.maxContentLength", 4096000)
+      case Success(value) => cluster.maxContentLength(value)
+      case Failure(_) => cluster.maxContentLength(4096000)
     }
 
-    val maxWaitForConnection = config.getInt("core.connector.connectionPool.maxWaitForConnection")
-    if (maxWaitForConnection > 0) conf.addProperty("connectionPool.maxWaitForConnection", maxWaitForConnection)
-
     val reconnectInterval = config.getInt("core.connector.connectionPool.reconnectInterval")
-    if (reconnectInterval > 0) conf.addProperty("connectionPool.reconnectInterval", reconnectInterval)
+    if (reconnectInterval > 0) cluster.reconnectInterval(reconnectInterval)
 
     val connectionMinSize = config.getInt("core.connector.connectionPool.minSize")
-    if (connectionMinSize > 0) conf.addProperty("connectionPool.minSize", connectionMinSize)
+    if (connectionMinSize > 0) cluster.minConnectionPoolSize(connectionMinSize)
 
     val connectionMaxSize = config.getInt("core.connector.connectionPool.maxSize")
-    if (connectionMaxSize > 0) conf.addProperty("connectionPool.maxSize", connectionMaxSize)
+    if (connectionMaxSize > 0) cluster.maxConnectionPoolSize(connectionMaxSize)
 
     val nioPoolSize = config.getInt("core.connector.nioPoolSize")
-    if (nioPoolSize > 0) conf.addProperty("nioPoolSize", nioPoolSize)
+    if (nioPoolSize > 0) cluster.nioPoolSize(nioPoolSize)
 
     val workerPoolSize = config.getInt("core.connector.workerPoolSize")
-    if (workerPoolSize > 0) conf.addProperty("workerPoolSize", workerPoolSize)
+    if (workerPoolSize > 0) cluster.workerPoolSize(workerPoolSize)
+
+    val conf = new util.HashMap[String, AnyRef]()
+    conf.put("ioRegistries", config.getAnyRef("core.connector.serializer.config.ioRegistries").asInstanceOf[java.util.ArrayList[String]])
+    val serializer = new GraphBinaryMessageSerializerV1()
```

Comment:
Seems like correct. we can expect a performance improvement
> GraphBinary is a new binary serialization format from TinkerPop that supersedes Gryo and it will eventually also replace GraphSON. GraphBinary is language independent and has a low serialization overhead which results in an improved performance.

https://docs.janusgraph.org/changelog/#graphbinary-is-now-supported